### PR TITLE
feat(generativeai): Add new sample for text generation

### PR DESCRIPTION
**Pull request by msampathkumar**
feat(generativeai): Add new sample for text generation

* YouTube examples
* HTTP support examples
* Controlled generation updates

> Note2: Raise a fresh PR for cleaner commmit history @ https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12633

> Note1: I have merged the changes from main branch, to get a fix for a CICD bug. Changes related to blunderbuss is not part of this change.

## Description

Fixes [b/369301403](http://b/369301403)

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))

### CODE REVIEW

File: generative_ai/text_generation/gemini_describe_http_pdf_example.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+def generate_content() -> str:
+    # [START generativeaionvertexai_gemini_describe_http_pdf]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO (developer): update project id
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    model = GenerativeModel("gemini-1.5-flash-001")
```
Comment (gericdong) [inline generative_ai/text_generation/gemini_describe_http_pdf_example.py:27]: Should the model be 002?

File: generative_ai/text_generation/gemini_describe_http_video_example.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+def generate_content() -> str:
+    # [START generativeaionvertexai_gemini_describe_http_video]
+    import vertexai
+    from vertexai.generative_models import GenerativeModel, Part
+
+    # TODO (developer): update project id
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    model = GenerativeModel("gemini-1.5-flash-001")
```
Comment (gericdong) [inline generative_ai/text_generation/gemini_describe_http_video_example.py:27]: Should the model be 002?

File: generative_ai/controlled_generation/example_06.py [hunk 3]
```diff
@@ -43,12 +43,14 @@ def generate_content() -> str:
...
         [
+            # Cloud storage image object
             Part.from_uri(
                 "gs://cloud-samples-data/generative-ai/image/office-desk.jpeg",
                 "image/jpeg",
             ),
+            # HTTP image
             Part.from_uri(
-                "gs://cloud-samples-data/generative-ai/image/gardening-tools.jpeg",
+                "https://storage.cloud.google.com/cloud-samples-data/generative-ai/image/gardening-tools.jpeg",
```
Comment (gericdong) [inline generative_ai/controlled_generation/example_06.py:53]: We will also need to update the model to 002 for this sample.

File: generative_ai/text_generation/gemini_describe_http_pdf_example.py (hunk 1 above)
Comment (msampathkumar) [reply generative_ai/text_generation/gemini_describe_http_pdf_example.py:27]: Yes. Next commit will include changes for `002`.

File: generative_ai/text_generation/gemini_describe_http_video_example.py (hunk 2 above)
Comment (msampathkumar) [reply generative_ai/text_generation/gemini_describe_http_video_example.py:27]: Yes. Next commit will include changes for `002`.

**Comment by msampathkumar** [comment]
Error: `projects/python-docs-samples-tests/locations/us-central1/publishers/google/models/gemini-1.5-flash-002`

**Comment by msampathkumar** [comment]
Update: Waiting for the my team to update the 002 access

**Comment by msampathkumar** [comment]
Error for `test_config_response_schema6`

`google.api_core.exceptions.InvalidArgument: 400 Provided image is not valid.`

**Comment by msampathkumar** [comment]
Working on the fix now.

File: generative_ai/video/gemini_describe_http_video_example.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    model = GenerativeModel("gemini-1.5-flash-002")
+
+    contents = [
+        # Text prompt
+        "Describe this video.",
+        # Example video ad for Pixel 8
+        Part.from_uri(
+            "https://storage.googleapis.com/cloud-samples-data/generative-ai/video/pixel8.mp4",
+            "video/map4",
```
Comment (gericdong) [inline generative_ai/video/gemini_describe_http_video_example.py:35]: map4 > mp4

**Comment by msampathkumar** [comment]
CICD Error: 

```
google.api_core.exceptions.Forbidden: 403 GET https://storage.googleapis.com/storage/v1/b/prompt_optimizer_bucket

Permission 'storage.buckets.get' denied on resource (or it may not exist).
```

These is a hidden edge case with code samples. Working with @Sita04 to address to fix it.

**Comment by msampathkumar** [comment]
Bug Fix @ https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12630

**Comment by msampathkumar** [comment]
@gericdong - Just checking! CICD failed for `test_prompt_optimizer` which is un-related to this PR. So can I have approval for this PR ?

**Comment by msampathkumar** [comment]
Error: `500 Internal error encountered`  - This seem to be a temporary availability issues. Restarting the test pipeline

**Review by gericdong — APPROVED** [review]

**Pull request closed without merging**